pkgs/applications/office/ticktick/default.nix

```diff
@@ -0,0 +1,65 @@
+{ lib, fetchurl, stdenv, wrapGAppsHook, dpkg, autoPatchelfHook,
+  glibc, gcc-unwrapped, nss, libdrm, mesa, alsa-lib, xdg-utils, systemd }:
+let
```

Comment:
A better pattern would be to define `pname`, `version` and `src` directly in `stdenv.mkDerivation`, and to instead give it a function with an argument `finalAttrs` as argument, and then refer to `finalAttrs.pname` for example.

See https://nixos.org/manual/nixpkgs/stable/#mkderivation-recursive-attributes

Reply:
That doubles the length of a lot of my substitutions in the install phase, I feel it just makes it read a lot less cleanly.